### ISSUE TITLE
Prevent redrawing of SVG inside gridster item

### ISSUE THREAD

**Issue by Sheepzez** [ManifestWebDesign/angular-gridster#356]
What's the best way to make my SVG elements inside gridster items only be drawn once, while still being allowed to drag them around?

Currently my SVG charts are redrawn many times per second while dragging, this is causing performance issues.

Currently my gridster section look like:

``` html
<div gridster="gridsterOpts" ng-controller="draggableGridController">
            <ul>
                <li gridster-item="widget" ng-repeat="widget in widgets">
                    <div class="panel panel-default" >
                        <div class="panel-heading">
                            ...
                        </div>
                        <div class="panel-body"><svg id="{{widget.id}}" class="{{widget.type}}"></svg></div>
                        {{ drawGraph(widget.id, widget.dataset, widget.type) }}
                    </div>
                </li>
            </ul>
        </div>
```

Clearly I don't entirely understand what's going on, but I can tell my drawGraph function shouldn't be there. But I can't figure out any location that isn't either calling it never, or many times per second.

Any help is much appreciated.

**Comment by btm1** [comment]
+1

**Comment by Sheepzez** [comment]
I believe one way to solve this would be to have event handlers for 'gridster-item-initialized' and 'gridster-item-transition-end' that call the drawGraph function once per event.

**Comment by btm1** [comment]
yeah it's pretty strange that you're essentially using the curly braces to force redrawing of your graph over and over again with every digest cycle {{ drawGraph(widget.id, widget.dataset, widget.type) }} 

I wouldn't do that. There are events for gridster for resize and move etc... and if you really need to "watch" the dataset on the widget that you're supplying to the graph and redraw the graph when it changes (which is what it looks like you're trying to do) I would do this in the controller for the widget "draggableGridController" i.e.:

```
$scope.$watchCollection('widget.dataset',function(ov,nv){
drawGraph($scope.widget.id, $scope.widget.dataset, $scope.widget.type)
});
```

What's happening if you don't do this way is that you're abusing the curly braces to redraw your chart which happens with every digest cycle and as the widget moves or is resized the widget object changes the column properties and size properties for the widget which triggers the digest cycle which triggers your redraw function...

**Comment by Sheepzez** [comment]
The data in the graphs isn't changing at all. Ideally I wouldn't need to redraw the graph on dragging at all, only on resize.  It seems though that the graphs (that are drawn after initial page load) are all deleted when any one graph is dragged.

It looks like the gridster items are being reset back to the initial template on drag. Any idea how to prevent this?

I only want to redraw on resize (for performance reasons).
Thanks for the help @btm1.

**Comment by btm1** [comment]
I am not seeing that be the case personally... I think because you've done this:

```
<div class="panel-body"><svg id="{{widget.id}}" class="{{widget.type}}"></svg></div>
                        {{ drawGraph(widget.id, widget.dataset, widget.type) }}
```

You are forcing your graphs to redraw all of them when one thing in the scope changes and a digest cycle happens...
